zipp/__init__.py

```diff
@@ -313,6 +322,47 @@ def iterdir(self):
         subs = map(self._next, self.root.namelist())
         return filter(self._is_child, subs)
 
+    def match(self, path_pattern):
+        return pathlib.Path(self.at).match(path_pattern)
+
+    def is_symlink(self):
+        return False  # See #82102
+
+    @contextlib.contextmanager
+    def _scandir(self):  # Needed for glob
+        yield self.iterdir()
+
+    def _make_child_relpath(self, relpath):  # Needed for glob
+        return self / relpath
+
+    def glob(self, pattern):
+        if not pattern:
+            raise ValueError("Unacceptable pattern: {!r}".format(pattern))
+        path = pathlib.Path(self.at)
+        drv, root, pattern_parts = path._flavour.parse_parts((pattern,))
+        if drv or root:
+            raise NotImplementedError("Non-relative patterns are unsupported")
+        if pattern[-1] in (path._flavour.sep, path._flavour.altsep):
+            pattern_parts.append('')
+        selector = pathlib._make_selector(tuple(pattern_parts), path._flavour)
+        for p in selector.select_from(self):
+            yield p
+
+    def rglob(self, pattern):
+        path = pathlib.Path(self.at)
+        drv, root, pattern_parts = path._flavour.parse_parts((pattern,))
+        if drv or root:
+            raise NotImplementedError("Non-relative patterns are unsupported")
+        if pattern and pattern[-1] in (path._flavour.sep, path._flavour.altsep):
+            pattern_parts.append('')
+        selector = pathlib._make_selector(("**",) + tuple(pattern_parts), path._flavour)
+        for p in selector.select_from(self):
+            yield p
+
+    def relative_to(self, *other, **kwargs):
+        other = (pathlib.Path(each.at) for each in other)
+        return pathlib.Path(self.at).relative_to(*other, **kwargs)
```

Comment:
I'm not sure this adequately captures concerns about zip files. It seems to be only concerned with the relative paths between the inner paths. What if the two zip files are distinct? What if they're in different directories? Perhaps distinct zip files should be disallowed. What about the relative path to a non-zipfile; should that be allowed?

To make matters worse, a zip file need not be on a disk. What should be the result when comparing the relative paths between two zipfiles downloaded from the internet (where there's no anchor for the zip files to tell if they came from the same location)?

Reply:
In 25739bc968, I've rewritten this function as a simple posixpath operation, which serves two benefits:

1. It's no longer concerned with platform-specifics.
2. It allows comparison across zipfiles.

It also removes the **kwargs. If this method really should serve the same purpose as pathlib, with a boolean parameter for 'walk up', probably more thought will need to go into this interface.